src/build.rs

```diff
@@ -310,14 +317,15 @@ impl Build {
 
         if !self.disable_zig_linker {
             // setup zig as linker
-            if let Some(target) = self.target.as_ref() {
-                let rustc_meta = rustc_version::version_meta()?;
-                let host_target = &rustc_meta.host;
-                // we only setup zig as linker when target isn't exactly the same as host target
-                if host_target != target {
-                    if let Some(rust_target) = rust_target {
+            let rustc_meta = rustc_version::version_meta()?;
+            let host_target = &rustc_meta.host;
+            // we only setup zig as linker when target isn't exactly the same as host target
+
+            for (i, full_target) in self.target.iter().enumerate() {
+                if host_target != full_target {
+                    if let Some(rust_target) = self.target.get(i) {
                         let env_target = rust_target.to_uppercase().replace('-', "_");
-                        let (zig_cc, zig_cxx) = prepare_zig_linker(target)?;
+                        let (zig_cc, zig_cxx) = prepare_zig_linker(full_target)?;
                         if is_mingw_shell() {
                             let zig_cc = zig_cc.to_slash_lossy();
                             build.env("TARGET_CC", &zig_cc);
```

Comment:
`TARGET_CC` and `TARGET_CXX` may need to be changed to `CC_${env_target}` and `CXX_${env_target}` now.

Reply:
Ok, I'll look it up in a few hours when I'm back at work.